octobot_commons/databases/cache_manager.py

```diff
@@ -0,0 +1,223 @@
+#  Drakkar-Software OctoBot-Commons
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import os
+
+import octobot_commons.logging as logging
+import octobot_commons.databases.document_database_adaptors as adaptors
+import octobot_commons.databases.implementations.cache_timestamp_database as cache_timestamp_database
+import octobot_commons.constants as common_constants
+import octobot_commons.symbol_util as symbol_util
+import octobot_commons.errors as common_errors
+
+
+class CacheManager:
+    """
+    Manages cache as a global dict since caches can be accessed from live, backtesting and optimizers concurrently
+    """
+
+    CACHES = {}
+    DEFAULT_CONFIG_IDENTIFIER = "default"
+
+    def __init__(self, database_adaptor=adaptors.TinyDBAdaptor):
+        self.database_adaptor = database_adaptor
+        self.logger = logging.get_logger(self.__class__.__name__)
+
+    def get_cache(self, tentacle, tentacle_name, exchange_name, symbol, time_frame, config_name,
+                  tentacles_setup_config, tentacles_requirements,
+                  cache_type=cache_timestamp_database.CacheTimestampDatabase, open_if_missing=True) -> tuple:
+        identifier = config_name or self.DEFAULT_CONFIG_IDENTIFIER
+        try:
+            return self.__class__.CACHES[tentacle_name][exchange_name][symbol][time_frame][identifier].get_database()
```

Comment:
Why not using something with nodes like Matrix? This is exactly the same need (not having a big dict with keys that can be None)

Reply:
Yep, using `EventTree`

Reply:
I added `get_nested_children_with_path` and `get_children_keys` to `EventTree` to use it here